geomstats/_backend/pytorch/__init__.py

```diff
@@ -349,7 +349,27 @@ def trace(x):
 
 
 def linspace(start, stop, num=50, dtype=None):
-    return _torch.linspace(start=start, end=stop, steps=num, dtype=dtype)
+    try:
```

Comment:
Can you please solve this with an `if` instead of `except`? i.e. you should know the cases in which it raises a `TypeError`, so you manipulate the variables before (as you do in the `except` branch).